lib/net/tcp_client/tcp_client.rb

```diff
@@ -45,8 +45,14 @@ class TCPClient
     #   localhost:2000
     attr_reader :server
 
-    attr_accessor :read_timeout, :connect_timeout, :connect_retry_count,
-      :retry_count, :connect_retry_interval, :server_selector, :close_on_error
+    attr_accessor :read_timeout, :write_timeout, :connect_timeout,
+      :connect_retry_count, :retry_count, :connect_retry_interval,
+      :server_selector, :close_on_error, :ssl_connect_timeout,
+      :ssl_context_params, :ssl_connect_retry_count,
+      :ssl_connect_retry_interval
```

Comment:
Can we drop the number of config options by rolling `:ssl_connect_timeout` into `:connect_timeout`, with the new `:connect_timeout` including the time for both the socket connect and the time to do the SSL handshake/connect?

Similarly for `:ssl_connect_retry_count` and `:ssl_connect_retry_interval`, could we just use the non-ssl equivalents?